src/StringManipulation.cpp

```diff
@@ -48,33 +48,33 @@ namespace Mezzanine
 {
     namespace Testing
     {
-        Mezzanine::String RightTrim(const Mezzanine::String& Text)
+        Mezzanine::String RightTrim(const Mezzanine::StringView Text)
         {
             Mezzanine::String HayStack(Text);
-            Mezzanine::String::size_type found{HayStack.find_last_not_of(" \n\r\t")};
+            Mezzanine::StringView::size_type found{HayStack.find_last_not_of(" \n\r\t")};
             if (found != String::npos)
                 { HayStack.erase(found+1); }
             else
                 { HayStack.clear(); } // HayStack is all whitespace
             return HayStack;
         }
```

Comment:
Wow! This implementation is suboptimal. This should be capturing the last character then only copy what is needed instead of copying the whole StringView.

Reply:
This is probably my fault.

Reply:
I don't think I wrote that.  I want to declare that out of scope for the PR.

Reply:
I will create an issue. This might be something a newb can do or we can do on a slow day.